### PR TITLE
[stdlib] Implement `join()` with variadic params in string literals

### PR DESCRIPTION
**Pull request by msaelices**
To be consistent with the `String.join()` signatures

### CODE REVIEW

File: stdlib/src/builtin/string_literal.mojo [hunk 1]
```diff
@@ -417,6 +417,51 @@ struct StringLiteral(
...
+        @parameter
+        fn add_elt[T: Stringable](a: T):
+            if is_first:
+                is_first = False
+            else:
+                result += self
+            result += str(a)
+
+        elems.each[add_elt]()
+        _ = is_first
```
Comment (JoeLoser) [inline stdlib/src/builtin/string_literal.mojo:462]: **Question** Is this needed?

**Review by JoeLoser — APPROVED** [review]

File: stdlib/src/builtin/string_literal.mojo (hunk 1 above)
Comment (msaelices) [reply stdlib/src/builtin/string_literal.mojo]: Actually not, removed and also for String: https://github.com/msaelices/mojo/commit/50767b081962027c388741092e07c04628ca96c6

Comment (msaelices) [reply stdlib/src/builtin/string_literal.mojo]: @JoeLoser, I am pinging you just to say that I've solved the conflicts. Thanks in advance

**Review by JoeLoser — APPROVED** [review]

**Comment by JoeLoser** [comment]
!sync

**Comment by JoeLoser** [comment]
FYI this is failing CI internally on both m7i and m7g instances, e.g.

```
source/mojo/stdlib/test/builtin/test_string_literal.mojo.test.runfiles/_main/open-source/mojo/stdlib/test/builtin/test_string_literal.mojo:224:17: AssertionError: `left == right` comparison failed:
   left: ,1,abc,3
  right: 1,abc,3
```

I'll rebase and see if it's still an issue, but I'm mildly surprised this code passed on the ubuntu runner on OSS CI but fails internally.

**Pull request closed without merging**